docs/source/features/huggingface_model_optimization.md

```diff
@@ -194,24 +194,22 @@ Pass config:
 
 Example: datasets in `data_configs`:
 ```json
-"data_configs": {
-    "oasst1_train": {
-        "name": "oasst1",
-        "type": "HuggingfaceContainer",
-        "params_config": {
-            "data_name": "timdettmers/openassistant-guanaco",
-            "split": "train",
-            "component_kwargs": {
-                "pre_process_data": {
-                    "text_cols": ["text"],
-                    "corpus_strategy": "line-by-line",
-                    "source_max_len": 512,
-                    "pad_to_max_len": false
-                }
+"data_configs": [{
+    "name": "oasst1_train",
```

Comment:
using the name instead of key would have the possibility that there are multiple dataset have the same name. we need dedup them in our code. 

For me, it seems we could leverage key

Reply:
> dedup them in our code

Yes, but this happens once during validation. Not necessarily a hot code path.

Having DataConfig::name makes it easier to identify object when debugging/printing. Having the key doesn't provide any such advantage.

Reply:
@jambayk, @trajepl, what's your opinion?

Reply:
Hitesh and I previously discussed the two options. I was originally for removing the name field and keeping the dict key. But the name for debugging purpose sounds useful too. 

I think it the decision depends on whether the debugging usefulness outweights the benefits of keeping the keys (1. consistent with other fields like systems, evaluators, etc, 2.guaranteed unique keys 3. direct index into data_configs)

Reply:
Thanks @jambayk, I will let you and @trajepl to make decision since I am ok with either approach.